### PR TITLE
Add a new weapon, the varmint air rifle

### PR DESCRIPTION
**Pull request by actually-the-artificer**
#### Summary
Content "Add a new weapon, the varmint air rifle."

#### Purpose of change
The BB gun is set to be changed as outlined in #63763. This PR brings back the damage-dealing air rifle with the introduction of the Varmint Air Rifle.

#### Describe the solution
Add the Varmint Air rifle, based off the .177 Ruger Targis High Velocity Air Rifle. It is best used for hunting small game, though can technically be pressed into combat if needed.

#### Describe alternatives you've considered
Not doing this, or waiting for someone else.

#### Testing
Loaded this up in game, tested it works.

### CODE REVIEW

**Comment by Maddremor** [comment]
Remove the clip size, that rifle is single shot. I would also look over the damage it does with the balance guidelines.

**Comment by actually-the-artificer** [comment]
Checked the balance guidelines and it says that .25 ACP does ~12 dmg?

**Comment by actually-the-artificer** [comment]
Just checked the balance doc and it says that .25 ACP deals 12 dmg? I think 4-6 is enough.

**Comment by Maddremor** [comment]
What is the bullet mass and muzzle velocity?

**Comment by Maleclypse** [comment]
Is this air rifle skipping multimag c++ needs because it’s a pump action that you air up one at a time to fire?

File: data/json/items/ranged/pneumatic.json [hunk 1]
```diff
@@ -36,5 +35,41 @@
...
+    "material": [ "plastic", "steel" ],
+    "flags": [ "NEVER_JAMS", "NON_FOULING" ],
+    "skill": "rifle",
+    "ammo": [ "airgun_pellet" ],
+    "weight": "3674 g",
+    "volume": "1500 ml",
+    "bashing": 9,
+    "to_hit": -1,
+    "range": 14,
+    "ranged_damage": { "damage_type": "stab", "amount": 2 },
```
Comment (Maddremor) [inline data/json/items/ranged/pneumatic.json:60]: Why is there stab damage here?

**Comment by Maddremor** [comment]
> Is this air rifle skipping multimag c++ needs because it’s a pump action that you air up one at a time to fire?

This is what I assume, and what is needed for the PR to get merged as far as I can tell.

There needs to be some kind of research for how long it takes to pump it, and how strong you have to be to do so.

**Comment by kevingranade** [comment]
This should represent a typical break action rifle,  not what seems to be a decided outlier. In particular the accessory rails and supressor are AFAICT very rare.
At a glance the stats look mostly fine, though peak damage around 4 is about right, along with increasing range to about 30.
The real issue here is going to be spawn frequency, these aren't remotely as common as hunting firearms.

File: data/json/items/ranged/pneumatic.json (hunk 1 above)
Comment (actually-the-artificer) [reply data/json/items/ranged/pneumatic.json:60]: Hrm, you're probs right. I didn't check the "normal" damage for an air rifle or the average barrel length.

**Comment by actually-the-artificer** [comment]
> This should represent a typical break action rifle, not what seems to be a decided outlier. In particular the accessory rails and supressor are AFAICT very rare. At a glance the stats look mostly fine, though peak damage around 4 is about right, along with increasing range to about 30. The real issue here is going to be spawn frequency, these aren't remotely as common as hunting firearms.

Seems like I will have to choose a different model for the varmint air rifle. Last night I was trying to find stats for air rifle commonality without too much success.

File: data/json/items/ranged/pneumatic.json [hunk 2]
```diff
@@ -15,9 +15,8 @@
     "flags": [ "NEVER_JAMS", "NON_FOULING" ],
     "skill": "rifle",
     "ammo": [ "BB" ],
-    "weight": "1000 g",
+    "weight": "1723 g",
     "volume": "1500 ml",
-    "longest_side": "90 cm",
```
Comment (flybye3o6) [inline data/json/items/ranged/pneumatic.json:20]: dont forget to remove the changes to "bbgun" in pneumatic

Comment (actually-the-artificer) [reply data/json/items/ranged/pneumatic.json:20]: What'd you mean by this?

**Comment by actually-the-artificer** [comment]
@Maddremor can you give this a quick look? I finally have a bit of time to work on it.

File: data/json/items/ammo.json [hunk 3]
```diff
@@ -1543,5 +1543,47 @@
     "flags": [ "TRADER_AVOID" ],
     "phase": "liquid",
     "ammo_type": "red_pen_ink"
+  },
+  {
+    "type": "AMMO",
+    "id": "airgun_pellet_target",
+    "price": 500,
+    "price_postapoc": 50,
+    "name": { "str": "air rifle target pellet", "str_pl": "air rifle target pellets" },
```
Comment (TheShadowFerret) [inline data/json/items/ammo.json:1552]:
```suggestion
    "name": { "str": "air rifle target pellet" },
```

File: data/json/items/ammo.json [hunk 4]
```diff
@@ -1543,5 +1543,47 @@
...
+    "stack_size": 200,
+    "loudness": 9,
+    "effects": [ "NOGIB", "NEVER_MISFIRES", "NON_FOULING" ]
+  },
+  {
+    "type": "AMMO",
+    "id": "airgun_pellet_hunting",
+    "price": 500,
+    "price_postapoc": 50,
+    "name": { "str": "air rifle hunting pellet", "str_pl": "air rifle hunting pellets" },
```
Comment (TheShadowFerret) [inline data/json/items/ammo.json:1573]:
```suggestion
    "name": { "str": "air rifle hunting pellet" },
```

**Comment by YourCompanionCube** [comment]
> This should represent a typical break action rifle, not what seems to be a decided outlier. In particular the accessory rails and supressor are AFAICT very rare. At a glance the stats look mostly fine, though peak damage around 4 is about right, along with increasing range to about 30. The real issue here is going to be spawn frequency, these aren't remotely as common as hunting firearms.

I agree about the accessories (though a number come with basic scopes) but as for their commonness you can find several at literally any walmart in the sporting goods section.

I own one that's a single shot break action powered by what is effectively a spring (it compresses air in a piston when you break it open, which then acts as a spring) and has a velocity of ~1250 ft/s with a 0.5 gram (7.8 grain) pellet which comes out to 36.2 joules which would be 6 damage. This doesn't seem out of line to me as these are basically like firing .22 short. (and is loud enough that hearing protection is maybe a good idea if you're shooting a lot, though still much quieter than an actual gun)

The reload should take more than 1 second since cocking it takes a fair bit of force, but it shouldn't be anywhere near as long as a crossbow. I'd say with inserting the pellet it should take maybe 1.5-2.5 seconds?

**Comment by Maleclypse** [comment]
Can you resolve conflicts?

File: data/json/items/ranged/pneumatic.json [hunk 5]
```diff
@@ -34,5 +34,43 @@
...
+    "price_postapoc": 550,
+    "//": "Based on the Gamo Varmint .177, with 1200fps PBA/1000fps Lead",
+    "material": [ "plastic", "steel" ],
+    "flags": [ "NEVER_JAMS", "NON_FOULING" ],
+    "skill": "rifle",
+    "ammo": [ "airgun_pellet" ],
+    "weight": "3243 g",
+    "volume": "1500 ml",
+    "longest_side": "85 cm",
+    "bashing": 9,
```
Comment (Maleclypse) [inline data/json/items/ranged/pneumatic.json:57]:
```suggestion
```

File: data/mods/Generic_Guns/ammo/rifle_pellet.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+[
+  {
+    "type": "AMMO",
+    "id": "rifle_pellet_target",
+    "copy-from": "airgun_pellet_target",
+    "name": { "str": "air rifle target pellet", "str_pl": "air rifle target pellets" },
```
Comment (Maleclypse) [inline data/mods/Generic_Guns/ammo/rifle_pellet.json:6]:
```suggestion
    "name": { "str": "air rifle target pellet" },
```

File: data/mods/Generic_Guns/ammo/rifle_pellet.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    "description": "A tin of small cone-shaped pellets meant to be used in an air rifle.  Suitable for plinking."
+  },
+  {
+    "type": "AMMO",
+    "id": "rifle_pellet_hunting",
+    "copy-from": "airgun_pellet_hunting",
+    "name": { "str": "air rifle hunting pellet", "str_pl": "air rifle hunting pellets" },
```
Comment (Maleclypse) [inline data/mods/Generic_Guns/ammo/rifle_pellet.json:13]:
```suggestion
    "name": { "str": "air rifle hunting pellet" },
```

**Comment by Maleclypse** [comment]
``` (continued from above) ERROR : src/item.cpp:741 [item &item::ammo_set(const itype_id &, int)] Tried to set invalid ammo of rifle_pellet_target for pellet_airgun
(all_mods)=> 17:42:10.075 ERROR : src/item_factory.cpp:2407 [void Item_factory::check_definitions() const] warnings for type pellet_airgun:
(all_mods)=> there is no actual ammo of type ammo_rifle_pellet defined
(all_mods)=> 
(all_mods)=> 17:42:12.579 ERROR : src/item.cpp:741 [item &item::ammo_set(const itype_id &, int)] Tried to set invalid ammo of rifle_pellet_target for pellet_airgun
```

File: data/mods/Generic_Guns/firearms/pneumatic.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+[
+  {
+    "id": "pellet_airgun",
+    "copy-from": "varmint_airgun",
+    "type": "GUN",
+    "name": { "str": "varmint air rifle" },
+    "description": "A cheap varmint air rifle chambered in .177, suitable for hunting small game.",
+    "ammo": [ "ammo_rifle_pellet" ],
```
Comment (Maleclypse) [inline data/mods/Generic_Guns/firearms/pneumatic.json:8]:
```suggestion
    "ammo": [ "airgun_pellet" ],
```

**Pull request merged**